recipes/nmos-cpp/all/conanfile.py

```diff
@@ -0,0 +1,243 @@
+    }
+    # "fPIC" is handled automatically by Conan, injecting CMAKE_POSITION_INDEPENDENT_CODE
+    default_options = {
+        "fPIC": True,
+        "with_dnssd": "mdnsresponder",
+    }
+
+    # wrapper CMakeLists.txt to call conan_basic_setup()
+    exports_sources = ["CMakeLists.txt"]
+    # use cmake_find_package_multi and prefer config-file packages
+    generators = "cmake", "cmake_find_package_multi"
+
+    short_paths = True
+
+    _cmake = None
+
+    # for out-of-source build, cf. wrapper CMakeLists.txt
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+        if self.settings.os == "Macos":
+            del self.options.with_dnssd
+        elif self.settings.os == "Linux":
+            self.options.with_dnssd = "avahi"
+        elif self.settings.os == "Windows":
+            self.options.with_dnssd = "mdnsresponder"
+
+    def requirements(self):
+        # for now, consistent with project's conanfile.txt
+        self.requires("boost/1.76.0")
+        self.requires("cpprestsdk/2.10.18")
+        self.requires("websocketpp/0.8.2")
+        self.requires("openssl/1.1.1k")
+        self.requires("json-schema-validator/2.1.0")
+        if self.options.get_safe("with_dnssd") == "mdnsresponder":
+            self.requires("mdnsresponder/878.200.35")
+            #self.options["mdnsresponder"].with_opt_patches = True
+        elif self.options.get_safe("with_dnssd") == "avahi":
+            self.requires("avahi/0.8")
+
+    def build_requirements(self):
+        # nmos-cpp needs CMake 3.17 or higher but CCI doesn't allow version ranges
+        self.build_requires("cmake/3.21.1")
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 11)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        # prefer config-file packages created by cmake_find_package_multi
+        # over any system-installed find-module packages
+        self._cmake.definitions["CMAKE_FIND_PACKAGE_PREFER_CONFIG"] = True
+        # (on Linux) select Avahi or mDNSResponder
+        self._cmake.definitions["NMOS_CPP_USE_AVAHI"] = self.options.get_safe("with_dnssd") == "avahi"
+        # (on Windows) use the Conan package for DNSSD (mdnsresponder), not the project's own DLL stub library
+        self._cmake.definitions["NMOS_CPP_USE_BONJOUR_SDK"] = True
+        # no need to build unit tests
+        self._cmake.definitions["NMOS_CPP_BUILD_TESTS"] = False
+        # the examples (nmos-cpp-registry and nmos-cpp-node) are useful utilities for users
+        self._cmake.definitions["NMOS_CPP_BUILD_EXAMPLES"] = True
```

Comment:
```suggestion
        self._cmake.definitions["NMOS_CPP_BUILD_EXAMPLES"] = False
```

Are these tools that consumer projects will use or just examples? If these are just examples, better to leave the users decide if they want to build them (not by default in ConanCenter).

Reply:
👍 These are useful utilities for users (see comment on previous line)

Reply:
Then why not using `NMOS_CPP_BUILD_UTILITIES` to activate/deactivate it? This is a suggestion for the library itself.

Reply:
Yes, could have been `BUILD_APPS`, `BUILD_EXECUTABLES`, or something, but `BUILD_EXAMPLES` won out at the time...